tests/gtest/avifstreamtest.cc

```diff
@@ -0,0 +1,127 @@
+// Copyright 2023 Google LLC
+// SPDX-License-Identifier: BSD-2-Clause
+
+#include <algorithm>
+#include <cstdint>
+#include <cstring>
+#include <vector>
+
+#include "avif/internal.h"
+#include "aviftest_helpers.h"
+#include "gtest/gtest.h"
+
+namespace libavif {
+namespace {
+
+//------------------------------------------------------------------------------
+
+TEST(StreamTest, Roundtrip) {
+  // TODO(yguyon): Check return values once pull request #1498 is merged.
+
+  // Write some fields.
+  testutil::AvifRwData rw_data;
+  avifRWStream rw_stream;
+  avifRWStreamStart(&rw_stream, &rw_data);
+  EXPECT_EQ(avifRWStreamOffset(&rw_stream), size_t{0});
+
+  const uint8_t rw_somedata[] = {3, 1, 4};
+  avifRWStreamWrite(&rw_stream, rw_somedata, sizeof(rw_somedata));
+
+  const char rw_somechars[] = "somechars";
+  avifRWStreamWriteChars(&rw_stream, rw_somechars,
+                         std::strlen(rw_somechars) + 1);
```

Comment:
Nit: We can replace `std::strlen(rw_somechars) + 1` with `sizeof(rw_somechars)`.